go/libraries/doltcore/env/actions/table.go

```diff
@@ -141,11 +141,18 @@ func GetRemoteBranchRef(ctx context.Context, ddb *doltdb.DoltDB, name string) (r
 		return ref.RemoteRef{}, false, err
 	}
 
+	// This will only do this behavior of guessing the remote branch if there is exactly one remote tracking branch with the matching branch name.
+	var found = false
+	var remoteRef ref.RemoteRef
 	for _, rf := range refs {
 		if remRef, ok := rf.(ref.RemoteRef); ok && remRef.GetBranch() == name {
-			return remRef, true, nil
+			if found {
+				return ref.RemoteRef{}, false, nil
```

Comment:
Nice job adding this in to resolve any ambiguity when there are multiple remote tracking branches that match the new branch name. 

It would be nice to return an error message in this case, to match Git's behavior. Here's what Git sends back: 
```
hint: If you meant to check out a remote tracking branch on, e.g. 'origin',
hint: you can do so by fully qualifying the name with the --track option:
hint: 
hint:     git checkout --track origin/<name>
hint: 
hint: If you'd like to always have checkouts of an ambiguous <name> prefer
hint: one remote, e.g. the 'origin' remote, consider setting
hint: checkout.defaultRemote=origin in your config.
fatal: 'branch10' matched multiple (2) remote tracking branches
```

I think there are two good options for updating this method... 
1 - since this is only used by checkout, you could directly return an error, probably just the last line of the error message above, so people know why checkout failed, although if you can include a hint, that's even better of course. :-) 
2 – alternatively, you could have this method return all the matching remote tracking branches and have both callers check that only one is returned, and then send that error. 

Either way, it would be good to update the method comments to mention that aspect of the method's contract.